### PR TITLE
Test/add more controller integration tests

### PR DESCRIPTION
**Pull request by andreamangion**

### CODE REVIEW

**Comment by andreamangion** [comment]
SEE THIS PULL REQUEST THIRD!

File: continuous-testing-foundations/dotnet/Pragmatest.Wallets/test/Pragmatest.Wallets.Web.UnitTests/WalletControllerUnitTests.cs [hunk 1]
```diff
@@ -39,14 +42,134 @@ public async Task Balance_GetBalanceIs4_ReturnsOkAndBalance4()
...
+
+            Deposit deposit = new Deposit { Amount = depositAmount };
+
+            walletServiceMock
+                .Setup(walletService => walletService.DepositFundsAsync(deposit))
+                .Returns(Task.FromResult(expectedBalance));
+
+            IWalletService walletService = walletServiceMock.Object;
+
+            WithdrawalRequest depositRequest = new WithdrawalRequest { Amount = depositAmount };
```
Comment (clydevassallo) [inline continuous-testing-foundations/dotnet/Pragmatest.Wallets/test/Pragmatest.Wallets.Web.UnitTests/WalletControllerUnitTests.cs]: I think this is meant to be a DepositRequest object

File: continuous-testing-foundations/dotnet/Pragmatest.Wallets/test/Pragmatest.Wallets.Web.UnitTests/WalletControllerUnitTests.cs [hunk 2]
```diff
@@ -39,14 +42,134 @@ public async Task Balance_GetBalanceIs4_ReturnsOkAndBalance4()
...
+            // Arrange
+            ILogger<WalletController> logger = Mock.Of<ILogger<WalletController>>();
+
+            Mock<IWalletService> walletServiceMock = new Mock<IWalletService>();
+
+            Deposit deposit = new Deposit { Amount = depositAmount };
+
+            IWalletService walletService = walletServiceMock.Object;
+
+            WithdrawalRequest depositRequest = new WithdrawalRequest { Amount = depositAmount };
```
Comment (clydevassallo) [inline continuous-testing-foundations/dotnet/Pragmatest.Wallets/test/Pragmatest.Wallets.Web.UnitTests/WalletControllerUnitTests.cs]: Same here

File: continuous-testing-foundations/dotnet/Pragmatest.Wallets/src/Pragmatest.Wallets.Web/Controllers/WalletController.cs [hunk 3]
```diff
@@ -34,10 +34,15 @@ public async Task<ActionResult<BalanceResponse>> Balance()
         }
 
         [HttpPost]
-        public async Task<IActionResult> Deposit(DepositRequest depositRequest)
+        public async Task<ActionResult<BalanceResponse>> Deposit(WithdrawalRequest depositRequest)
```
Comment (clydevassallo) [inline continuous-testing-foundations/dotnet/Pragmatest.Wallets/src/Pragmatest.Wallets.Web/Controllers/WalletController.cs]: Why is the deposit endpoint taking a WithdrawalRequest?

File: continuous-testing-foundations/dotnet/Pragmatest.Wallets/src/Pragmatest.Wallets.Web/Mappers/WalletMappingProfile.cs [hunk 4]
```diff
@@ -9,7 +9,7 @@ public class WalletMappingProfile : Profile
         public WalletMappingProfile()
         {
             CreateMap<Balance, BalanceResponse>();
-            CreateMap<DepositRequest, Deposit>();
+            CreateMap<WithdrawalRequest, Deposit>();
```
Comment (clydevassallo) [inline continuous-testing-foundations/dotnet/Pragmatest.Wallets/src/Pragmatest.Wallets.Web/Mappers/WalletMappingProfile.cs]: I think this change is related to the others but is a mistake

**Review by clydevassallo — CHANGES_REQUESTED** [review]

File: continuous-testing-foundations/dotnet/Pragmatest.Wallets/test/Pragmatest.Wallets.Web.UnitTests/WalletControllerUnitTests.cs (hunk 1 above)
Comment (andreamangion) [reply continuous-testing-foundations/dotnet/Pragmatest.Wallets/test/Pragmatest.Wallets.Web.UnitTests/WalletControllerUnitTests.cs]: Fixed

File: continuous-testing-foundations/dotnet/Pragmatest.Wallets/test/Pragmatest.Wallets.Web.UnitTests/WalletControllerUnitTests.cs (hunk 2 above)
Comment (andreamangion) [reply continuous-testing-foundations/dotnet/Pragmatest.Wallets/test/Pragmatest.Wallets.Web.UnitTests/WalletControllerUnitTests.cs]: Fixed

File: continuous-testing-foundations/dotnet/Pragmatest.Wallets/src/Pragmatest.Wallets.Web/Controllers/WalletController.cs (hunk 3 above)
Comment (andreamangion) [reply continuous-testing-foundations/dotnet/Pragmatest.Wallets/src/Pragmatest.Wallets.Web/Controllers/WalletController.cs]: Fixed

File: continuous-testing-foundations/dotnet/Pragmatest.Wallets/src/Pragmatest.Wallets.Web/Mappers/WalletMappingProfile.cs (hunk 4 above)
Comment (andreamangion) [reply continuous-testing-foundations/dotnet/Pragmatest.Wallets/src/Pragmatest.Wallets.Web/Mappers/WalletMappingProfile.cs]: Fixed

**Review by clydevassallo — APPROVED** [review]

**Pull request merged**